ios/brave-ios/Sources/BraveWallet/Crypto/Stores/AccountActivityStore.swift

```diff
@@ -223,10 +226,25 @@ class AccountActivityStore: ObservableObject, WalletObserverStore {
       )
 
       self.isLoadingAccountFiat = true
-      let tokenBalances = await self.rpcService.fetchBalancesForTokens(
-        account: account,
-        networkAssets: allUserNetworkAssets
-      )
+      var tokenBalances: [String: Double] = [:]
+      if account.coin == .btc {
+        let networkAsset = allUserNetworkAssets.first {
+          $0.network.supportedKeyrings.contains(account.keyringId.rawValue as NSNumber)
+        }
+        if let btc = networkAsset?.tokens.first,
+          let btcBalance = await self.bitcoinWalletService.fetchBTCBalance(
+            accountId: account.accountId
+          )
+        {
+          tokenBalances = [btc.id: btcBalance]
+        }
+      } else {
+        tokenBalances = await self.rpcService.fetchBalancesForTokens(
+          account: account,
+          networkAssets: allUserNetworkAssets
+        )
+        tokenBalanceCache.merge(with: tokenBalances)
+      }
```

Comment:
This will be cleaner after [#36764](https://github.com/brave/brave-browser/issues/36764). We would just be fetching balance from `WalletUserAssetBalance` / our CoreData cache model here so we shouldn't need separate BTC logic. I'm not sure it warrants a TODO comment & linking to issue as the issue itself already states where changes need made. Just noting for other code reviewers.